geoalchemy2/types/__init__.py

```diff
@@ -179,7 +179,8 @@ def process(bindvalue):
     @staticmethod
     def check_ctor_args(geometry_type, srid, dimension, use_typmod, nullable):
         try:
-            srid = int(srid)
+            # passing default SRID is it is NULL from DB
+            srid = int(srid or -1)
```

Comment:
```suggestion
            srid = int(srid if srid is not None else -1)
```
To avoid changing `0` into `-1`

Reply:
I have fixed  this change in the commit